File: lang/php/test/vac.fun/algorithms/meta/SortingTest.php
<?php
    require_once "vac.fun/algorithms/meta/Sorting.php";
    require_once "vac.fun/util/SequenceBuilder.php";
    require_once "vac.fun/util/Sequences.php";
    require_once "vac.fun/util/TestRunner.php";

    use vac\fun\algorithms\meta\Sorting;
    use vac\fun\util\SequenceBuilder;
    use vac\fun\util\Sequences;
    use vac\fun\util\TestRunner;

    function testBubbleSort(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packRandom($arr);

        $checksum = Sequences\parityChecksum($arr);

        Sorting\bubbleSort($arr);

        if (Sequences\parityChecksum($arr) != $checksum) {
            return false;
        }

        return Sequences\isSorted($arr);
    }

    function testInsertionSort(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packRandom($arr);

        $checksum = Sequences\parityChecksum($arr);

        Sorting\insertionSort($arr);

        if (Sequences\parityChecksum($arr) != $checksum) {
            return false;
        }

        return Sequences\isSorted($arr);
    }

    function testMergeSort(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packRandom($arr);

        $checksum = Sequences\parityChecksum($arr);

        Sorting\mergeSort($arr);

        if (Sequences\parityChecksum($arr) != $checksum) {
            return false;
        }

        return Sequences\isSorted($arr);
    }

    function testQuickSort(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packRandom($arr);

        $checksum = Sequences\parityChecksum($arr);

        Sorting\quickSort($arr);

        if (Sequences\parityChecksum($arr) != $checksum) {
            return false;
        }

        return Sequences\isSorted($arr);
    }

    function testSelectionSort(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packRandom($arr);

        $checksum = Sequences\parityChecksum($arr);

        Sorting\selectionSort($arr);

        if (Sequences\parityChecksum($arr) != $checksum) {
            return false;
        }

        return Sequences\isSorted($arr);
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\parseTest(testBubbleSort());

        TestRunner\parseTest(testInsertionSort());

        TestRunner\parseTest(testMergeSort());

        TestRunner\parseTest(testQuickSort());

        TestRunner\parseTest(testSelectionSort());
    }
?>

File: go/muse/algorithms/meta/Sorting/Sorting_test.go
package Sorting

import (
	"testing"

	"muse/util/SequenceBuilder"
	"muse/util/Sequences"
)

func TestBubbleSort(t *testing.T) {
	size := 32768

	arr := make([]int, size)
	SequenceBuilder.PackRandom(arr)

	checksum := Sequences.ParityChecksum(arr)

	BubbleSort(arr)

	if Sequences.ParityChecksum(arr) != checksum {
		t.FailNow()
	}

	if !Sequences.IsSorted(arr) {
		t.FailNow()
	}
}

func TestInsertionSort(t *testing.T) {
	size := 32768

	arr := make([]int, size)
	SequenceBuilder.PackRandom(arr)

	checksum := Sequences.ParityChecksum(arr)

	InsertionSort(arr)

	if Sequences.ParityChecksum(arr) != checksum {
		t.FailNow()
	}

	if !Sequences.IsSorted(arr) {
		t.FailNow()
	}
}

func TestMergeSort(t *testing.T) {
	size := 32768

	arr := make([]int, size)
	SequenceBuilder.PackRandom(arr)

	checksum := Sequences.ParityChecksum(arr)

	MergeSort(arr)

	if Sequences.ParityChecksum(arr) != checksum {
		t.FailNow()
	}

	if !Sequences.IsSorted(arr) {
		t.FailNow()
	}
}

func TestQuickSort(t *testing.T) {
	size := 32768

	arr := make([]int, size)
	SequenceBuilder.PackRandom(arr)

	checksum := Sequences.ParityChecksum(arr)

	QuickSort(arr)

	if Sequences.ParityChecksum(arr) != checksum {
		t.FailNow()
	}

	if !Sequences.IsSorted(arr) {
		t.FailNow()
	}
}

func TestSelectionSort(t *testing.T) {
	size := 32768

	arr := make([]int, size)
	SequenceBuilder.PackRandom(arr)

	checksum := Sequences.ParityChecksum(arr)

	SelectionSort(arr)

	if Sequences.ParityChecksum(arr) != checksum {
		t.FailNow()
	}

	if !Sequences.IsSorted(arr) {
		t.FailNow()
	}
}
